File: ios/RuterShortcutWidget/TransportMode.swift
//
//  TransportMode.swift
//  Runner
//
//  Created by Ludvig Marcel Carlsen on 12/05/2023.
//

import Foundation
import SwiftUI


enum TransportMode: String, Codable {
    case foot = "foot"
    case rail = "rail"
    case bus = "bus"
    case coach = "coach"
    case tram = "tram"
    case metro = "metro"
    case water = "water"
    case air = "air"
    case lift = "lift"
    
    static var transportModeColors: [TransportMode: Color] = [
        .bus: Color.bus,
        .rail: Color.rail,
        .tram: Color.tram,
        .metro: Color.metro,
        .foot: Color.foot,
        .air: Color.air,
        .water: Color.water
    ]
}

extension Color {
    static let bus = Color(red: 231/255, green: 1/255, blue: 0)
    static let rail = Color(red: 34/255, green: 94/255, blue: 225/255)
    static let tram = Color(red: 13/255, green: 144/255, blue: 239/255)
    static let metro = Color(red: 237/255, green: 112/255, blue: 8/255)
    static let foot = Color(red: 82/255, green: 83/255, blue: 93/255)
    static let air = Color(.gray)
    static let water = Color(.purple)
}

File: android/app/src/main/java/com/example/retur/TransportMode.kt
package com.example.retur

import android.graphics.Color

enum class TransportMode(val displayName: String) {
    bus("bus"),
    coach("coach"),
    tram("tram"),
    rail("train"),
    metro("metro"),
    water("ferry"),
    air("airplane"),
    lift("lift"),
    foot("foot"),
    unknown("unknown");

    companion object {
        fun fromJson(s: String): TransportMode {
            return valueOf(s)
        }

        fun getColor(mode: String): Int {
            return transportColorMap[mode] ?: Color.rgb(148, 148, 148)
        }
    }
}

private val transportColorMap = mapOf(
        TransportMode.bus.name to Color.rgb(230, 0, 0),
        TransportMode.tram.name to Color.rgb(11, 145, 239),
        TransportMode.rail.name to Color.rgb(0, 48, 135),
        TransportMode.metro.name to Color.rgb(236, 112, 12),
        TransportMode.water.name to Color.rgb(104, 44, 136),
        TransportMode.foot.name to Color.rgb(82, 83, 93)
)

private val transportAssetMap = mapOf(
        TransportMode.bus.name to "assets/bus.svg",
        TransportMode.coach.name to "assets/bus.svg",
        TransportMode.tram.name to "assets/tram.svg",
        TransportMode.rail.name to "assets/rail.svg",
        TransportMode.metro.name to "assets/metro.svg",
        TransportMode.water.name to "assets/water.svg",
        TransportMode.air.name to "assets/air.svg",
        TransportMode.foot.name to "assets/foot.svg"
)
